Watch Extension/MRExtensionDelegate.swift

```diff
@@ -87,6 +90,37 @@ class MRExtensionDelegate : NSObject, WKExtensionDelegate, MKMetadataConnectivit
 
     func metadataConnectivityDidReceiveExerciseModelMetadata(exerciseModelMetadata: [MKExerciseModelMetadata]) {
         self.exerciseModelMetadata = exerciseModelMetadata
+        MRExtensionDelegate.saveModelsMetadata(exerciseModelMetadata)
+    }
+    
+    static func loadModelsMetadata() -> [MKExerciseModelMetadata] {
+        if let data = NSFileManager.defaultManager().contentsAtPath(MRExtensionDelegate.fileUrl) {
+            do {
+                let jsonObj = try NSJSONSerialization.JSONObjectWithData(data, options: NSJSONReadingOptions.AllowFragments) as? [String: String]
+                if let jsonObj = jsonObj {
+                    let models = jsonObj.map { (id, name) -> MKExerciseModelMetadata in return (id, name) }
+                    NSLog("Loaded \(models.count) models")
+                    return models
+                }
+            } catch let error {
+                NSLog("Error while deserializing models metadata : \(error)")
+            }
+        }
+        return []
+    }
+    
+    static func saveModelsMetadata(models: [MKExerciseModelMetadata]) {
```

Comment:
Here we're on the watch and it saves the models metadata in order to display then in the picker.
(It needs to be saved to work without phone connection - otherwise you only see the default hardcoded models).

Note that if we get rid of the model selection we can remove this code too.